x-pack/plugins/security_solution/public/resolver/store/middleware/related_events_fetcher.ts

```diff
@@ -69,19 +72,31 @@ export function RelatedEventsFetcher(
             eventCategory,
             cursor: result.nextEvent,
             nodeID,
+            dataRequestID,
           },
         });
       }
     }
-
+    const oldID = selectors.currentNodeEventsInCategoryRequestID(state);
```

Comment:
These two selectors are only ever used together. Maybe you could replace them with a single selector called  something like: `eventsInCategoryResultIsStale`